### PR TITLE
Set Cloud provider on bootstrap

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
## Description
Put cloud-provider in ClusterConfig, so it can get set on k8s bootstrap.

### CODE REVIEW

File: src/k8s/api/v1/cluster_config.go [hunk 1]
```diff
@@ -103,6 +104,12 @@ type MetricsServerConfig struct {
 
 func (c MetricsServerConfig) GetEnabled() bool { return getField(c.Enabled) }
 
+type CloudProviderConfig struct {
+	CloudProvider *string `json:"cloud-provider,omitempty" yaml:"cloud-provider,omitempty"`
+}
+
+func (c UserFacingClusterConfig) GetCloudProvider() string { return getField(c.CloudProvider) }
```
Comment (neoaggelos) [inline src/k8s/api/v1/cluster_config.go:111]: revert, this is not used

File: src/k8s/cmd/k8s/k8s_set.go [hunk 2]
```diff
@@ -177,6 +177,8 @@ func updateConfig(config *apiv1.UserFacingClusterConfig, arg string) error {
 			return fmt.Errorf("invalid boolean value for metrics-server.enabled: %w", err)
 		}
 		config.MetricsServer.Enabled = &v
+	case "cloud-provider":
+		config.CloudProvider = vals.Pointer(value)
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_set.go:181]: revert this, we only add this to API for now

**Review by neoaggelos — COMMENTED** [review]
Nice work, two quick notes

**Review by neoaggelos — APPROVED** [review]

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**